### ISSUE TITLE
Vernaculars for higher-level taxa

### ISSUE THREAD

**Issue by hyanwong** [OneZoom/tree-build#84]
I just realised, we might well want to be able to harvest vernaculars (but definitely not images) for entire clades, e.g. Platyrrhini-> "New World Monkeys". We could either expand the remit of the `get_wiki_images script clade` option to do this, or have a new utility. My preference is for the latter. We could even move some of the existing vernaculars code into this new utility, and have it as a dependency of `get_wiki_images.py`.

We shouldn't need to make any online API requests for this, I think. All the information is presumably in the wikidata JSON? That should make it possible to run the script on a vanilla OneZoom instance, having removed all the existing vernaculars, and see what it looks like.

**Comment by davidebbo** [comment]
Yes, I agree it would make sense to do this. Indeed, this should be all in the wikidata JSON, exactly as it is for leaf vernaculars.

Note that in your specific example of Platyrrhini, there does not appear to be a common taxon name in wikidata: https://www.wikidata.org/wiki/Q237157. Hopefully, this is an isolated lapse.

Would it make sense to entirely decouple vernacular harvesting (leaf and higher taxa) from the get_wiki_images utility? i.e. not have get_wiki_images do that at all, and rely on a separate utility that just does get_wiki_vernaculars (for both leaf and higher taxa).

**Comment by hyanwong** [comment]
> Note that in your specific example of Platyrrhini, there does not appear to be a common taxon name in wikidata: https://www.wikidata.org/wiki/Q237157. Hopefully, this is an isolated lapse.

Oh, how strange, I never checked! I might add that then (could be a good test)

> Would it make sense to entirely decouple vernacular harvesting (leaf and higher taxa) from the get_wiki_images utility? i.e. not have get_wiki_images do that at all, and rely on a separate utility that just does get_wiki_vernaculars (for both leaf and higher taxa).

That sounds like it might be a good idea, but I don't want to cause lots of extra work. How difficult would it be to disentangle them, do you think?

**Comment by hyanwong** [comment]
> > Note that in your specific example of Platyrrhini, there does not appear to be a common taxon name in wikidata: https://www.wikidata.org/wiki/Q237157. Hopefully, this is an isolated lapse.
> 
> Oh, how strange, I never checked! I might add that then (could be a good test)

I just added it. But from a quick look, it appears as if https://www.wikidata.org/wiki/Property:P1843 is often not set for higher level taxa, and the vernaculars are put in the "also known as" section at the top of the wikidata page (I assume this is encoded somewhere else). See also e.g. https://www.wikidata.org/wiki/Q36611

I think we just have to wait until more vernaculars gradually get added via `Property:P1843`: the "also know as" fields are not very reliable. It might therefore be that we won't get many high-level vernaculars from wikidata at the moment, but we have them from other sources, so maybe we don't care that wikidata isn't yet very useful for this.

**Comment by davidebbo** [comment]
Right, the 'also known as' has "gorilla genus" for English, which is not ideal. Presumably, the ideal string here is "Gorillas".
